peps/pep-0777.rst

```diff
@@ -266,6 +301,61 @@ build tag in the middle of the file name makes any extensions ambiguous (i.e.
 ``foo-0.3-py3-none-any-fancy_new_tag.whl`` would parse as the build tag being
 ``py3``). This limits changes to information stored in the wheel file name.
 
+Store the Wheel Major Version in the File Extension (``.whl2``)
+---------------------------------------------------------------
+Storing the wheel major version in the file extension has several nice
+advantages. For one, there is no need to introduce the `Wheel-Version`
+metadata field, since installers could simply filter based on file extension.
+This would also allow future side-by-side packages. However, changing the
+extension for wheels each major version has some downsides. First, the version
+stored in the ``WHEEL`` file must match the file extension, and this would need
+to be verified by installers. Additionally, many systems associate file type by
+file extension (e.g. executable associations, various web caching software),
+and these would need to be updated every version that is released. Furthermore,
+part of the brittleness of the current wheel specification is that so much
+metadata is stored in the filename. Filenames are not well suited to store
+structured data. Moving away from encoding information in the filename should
+be a goal of future wheel revisions.
+
+Another possibility is to use the file extension to encode the outer container
+format (i.e. a ZIP file containing ``.dist-info``) separate from the inner
+wheel version. However, this could lead to confusion over a difference in
+`Wheel-Version` vs the version in the extension. If an installer errors that
```

Comment:
The implication here being that the extension may not always need to track the wheel version number, if e.g. future extensions can be handled in way that's backward compatible with the outer container format.  Am I expressing that correctly?  I agree with the rejection of this point, but I feel like there's perhaps a logic step that could be strengthened.